src/Core/Parser/SyntaxTree/ViewHelperNode.php

```diff
@@ -82,7 +82,8 @@ public function getViewHelperClassName(): string
 
     /**
      * @internal only needed for compiling templates
-     * @return NodeInterface[]
+     * @return array<NodeInterface|int|float|string|bool> For simple values, an argument might also be scalar
```

Comment:
You can also use `scalar` here. It's a valid declaration for PHPDoc, PHPStan and PHPStorm. 
```suggestion
     * @return array<NodeInterface|scalar> For simple values, an argument might also be scalar
```

Reply:
adjusted in #1051